backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/calendar/C_Year_StepDef.java

```diff
@@ -52,10 +58,23 @@ public void load_C_Year(@NonNull final DataTable dataTable)
 		final List<Map<String, String>> tableRows = dataTable.asMaps(String.class, String.class);
 		for (final Map<String, String> tableRow : tableRows)
 		{
+
 			final String fiscalYear = DataTableUtil.extractStringForColumnName(tableRow, I_C_Year.COLUMNNAME_FiscalYear);
 
-			final I_C_Year yearRecord = queryBL.createQueryBuilder(I_C_Year.class)
-					.addEqualsFilter(I_C_Year.COLUMNNAME_FiscalYear, fiscalYear)
+			final IQueryBuilder<I_C_Year> query = queryBL.createQueryBuilder(I_C_Year.class)
+					.addOnlyActiveRecordsFilter()
+					.addEqualsFilter(I_C_Year.COLUMNNAME_FiscalYear, fiscalYear);
+
+			final String calendarIdentifier = DataTableUtil.extractStringOrNullForColumnName(tableRow, "OPT." + I_C_Year.COLUMNNAME_C_Calendar_ID + "." + TABLECOLUMN_IDENTIFIER);
+			if (Check.isNotBlank(calendarIdentifier))
+			{
+				final I_C_Calendar calendar = calendarTable.get(calendarIdentifier);
+				Check.assumeNotNull(calendar, "No calendar provided", calendar);
```

Comment:
The message is misleading. The calendar may exist, but calendarIdentifier does not match any of the known calendars.
Also, the `calendar` param does not have any placeholder in the message...
```suggestion
				Check.assumeNotNull(calendar, "No calendar found for identifier; {}", calendarIdentifier);
```